lib/wallet/src/Cardano/Wallet.hs

```diff
@@ -2087,10 +2087,8 @@ buildSignSubmitTransaction ti db@DBLayer{..} netLayer txLayer pwd walletId
             & throwWrappedErr wrapNetworkError
             & liftIO
 
-        slotToUTCTime slot
-            & interpretQuery (neverFails "slot is ahead of the node tip" ti)
-            & fmap (BuiltTx{..},)
-            & liftIO
+        utcTime <- translateSlotToUTC timeTranslation slot & either throwIO pure
```

Comment:
Could we keep using `TimeInterpreter` here to be consistent with the rest of the wallet?